### ISSUE TITLE
Does not work at all with a Hero 9 Black

### ISSUE THREAD

**Issue by desrod** [jschmid1/gopro_as_webcam_on_linux#27]
After trying 17 other projects, I was hopeful that this one would work. Unfortunately, it goes on the heap with the other 17. 

When I plug in my Hero9, it creates a device, usb0, and passing this in as `./gopro webcam -p usb`, fails to discover the GoPro IP (because it actually doesn't _have_ an IP, since it's not broadcasting as a network device, it's connected over USB), so that fails with the unhelpful message of: 

```
Discovered: usb0.
Using usb0 to discover the GOPRO_IP.
Could not automatically discover a valid GOPRO_IP.
```

Since you can't use USB _and_ network at the same time, these options would seem to be in conflict with each other. 

How does one stream video from this device, plugged in via USB, when it's not on the network? 

Likewise, since the device itself can't connect _**TO**_ a wireless network, it can only be an AP, how does one get this thing visible as a device that can connect to OBS?

**Comment by desrod** [comment]
I've managed to get this working, and wrote a detailed blog post describing the steps for anyone who may fall into the same gaps I did. Thanks!

**Comment by msonowal** [comment]
@desrod 
would you mid sharing it as mine hero 10 does not connect with all the tries

**Comment by IceCereal** [comment]
@msonowal In case anyone stumbles across this in the future (like I did), going through their profile led me to this:

https://blog.gnu-designs.com/howto-use-your-gopro-hero8-hero9-and-hero10-as-a-webcam-on-linux-over-usb-c-without-the-media-mod/

Archived link just in case it goes down: https://archive.ph/xAX3I

I don't know if it works though.

**Comment by codecat** [comment]
I know this is an old thread, but this post had the key thing I was missing of getting this to work. The network interface never had an IP address for me, so I had to make sure the network interface had DHCP enabled. I made `/etc/netplan/10-gopro.yaml` with the following contents:
```yaml
network:
  version: 2
  renderer: networkd
  ethernets:
    enx00000000: # Put your actual interface name here
      dhcp4: yes
```
Then running `netplan generate`, `netplan apply`, and it started working!
